qiskit/providers/aer/backends/aerbackend.py

```diff
@@ -331,53 +331,46 @@ def _format_results(output):
                     data[key] = format_save_type(val, save_types[key], save_subtypes[key])
         return Result.from_dict(output)
 
-    @staticmethod
-    def _circuit_optypes(circuit):
-        """Retur set of all operation names in a circuit"""
-        if not isinstance(circuit, QuantumCircuit):
-            return set()
-        optypes = set()
-        for instr, _, _ in circuit._data:
-            optypes.add(type(instr))
-        return optypes
-
     def _assemble(self, circuits, parameter_binds=None, **run_options):
         """Assemble one or more Qobj for running on the simulator"""
-        # Remove quantum error instructions from circuits and add to
-        # run option noise model
-        circuits, optypes, run_options = self._assemble_noise_model(circuits, **run_options)
-
-        # This conditional check can be removed when we remove passing
-        # qobj to run
         if isinstance(circuits, (QasmQobj, PulseQobj)):
             qobj = circuits
-        elif parameter_binds:
-            # Handle parameter binding
-            parameterizations = self._convert_binds(circuits, parameter_binds)
-            assemble_binds = []
-            assemble_binds.append({param: 1 for bind in parameter_binds for param in bind})
-
-            qobj = assemble(
-                compile_circuit(
-                    circuits,
-                    basis_gates=self.configuration().basis_gates,
-                    optypes=optypes),
-                self,
-                parameter_binds=assemble_binds,
-                parameterizations=parameterizations)
         else:
-            qobj = assemble(
-                compile_circuit(
+            # Generate optypes for circuit
+            # Generate opsets of instructions
+            if not isinstance(circuits, list):
+                circuits = [circuits]
```

Comment:
Same point as before about trying to allow arbitrary iterables - in particular, it's usually surprising if you can't use `tuple` as a collection for these purposes (_technically_ it's more efficient than `list` for a few things).